pkg/resource/stack/secrets.go

```diff
@@ -524,3 +524,68 @@ func (bd *cachingBatchDecrypter) DecryptValue(ctx context.Context, ciphertext st
 func (bd *cachingBatchDecrypter) BatchDecrypt(ctx context.Context, ciphertexts []string) ([]string, error) {
 	return bd.decrypter.BatchDecrypt(ctx, ciphertexts)
 }
+
+// BatchEncrypt is a helper function that encapsulates the pattern of checking if a secrets manager
+// supports batching, starting a batch encryption operation, calling a function with the encrypter,
+// and completing the batch operation.
+func BatchEncrypt[T any](
+	ctx context.Context,
+	sm secrets.Manager,
+	fn func(context.Context, config.Encrypter) (T, error),
+) (T, error) {
+	var zero T
+	var completeBatch CompleteCrypterBatch
+	enc := sm.Encrypter()
+
+	if batchingSecretsManager, ok := sm.(BatchingSecretsManager); ok {
+		enc, completeBatch = batchingSecretsManager.BeginBatchEncryption()
```

Comment:
is it okay to not call `completeBatch` in case of error? or should this be something like

```
defer func() { err = errors.Join(err, completeBatch(ctx)) }()
```